### PR TITLE
Added new props : autoFocusOption

### PR DESCRIPTION
**Pull request by CharlieBreval**
Hello,

Recently, I have been facing a problem : Using React Select without preselection of the first available (focusable) option.
This option didn't exist at all. The problem has been raised in multiples issues : 
#2848 

Behavior with this option is simple : 
1 : Click on the select : No option is focused
2 : Your user starts typing a word, no option is focused until the user do an action for that.

### CODE REVIEW

File: src/Select.js [hunk 1]
```diff
@@ -399,8 +402,14 @@ export default class Select extends Component<Props, State> {
     ) {
       const selectValue = cleanValue(nextProps.value);
       const menuOptions = this.buildMenuOptions(nextProps, selectValue);
-      const focusedValue = this.getNextFocusedValue(selectValue);
-      const focusedOption = this.getNextFocusedOption(menuOptions.focusable);
+      let focusedValue = this.getNextFocusedValue(selectValue);
+      let focusedOption = this.getNextFocusedOption(menuOptions.focusable);
+
+      if (nextProps.autoFocusFirstOption === false  && this.state.focusedOption === null) {
```
Comment (Blightwidow) [inline src/Select.js]: It is usually better to consult the state directly in the `this.setState()` method as it is asynchronous. Something like 

```js
this.setState((prevState) => ({
  focusedOption: nextProps.autoFocusFirstOption === false  && prevState.focusedOption === null ? null : focusedOption,
  ...
}));
```

File: src/Select.js [hunk 2]
```diff
@@ -82,6 +82,8 @@ export type Props = {
   'aria-labelledby'?: string,
   /* Focus the control when it is mounted */
   autoFocus?: boolean,
+  /* Focus first focusable option on menu open and when user is typing */
+  autoFocusFirstOption: boolean,
```
Comment (Blightwidow) [inline src/Select.js]: I would maybe see something more like `autoFocusOption` with a value of `null`, `'first'`, or `'last'` for more flexibility.

Comment (CharlieBreval) [reply src/Select.js]: I will rename the prop, but 'first' or 'last' is decided in another part of the library, not the concern of this option.

Comment (Blightwidow) [reply src/Select.js]: Ok fine by me then.

**Comment by danishmughal** [comment]
+1 for this, sorely needed.

**Comment by alisonhall** [comment]
Any idea when this will be merged and released?

**Comment by shepitko** [comment]
> Any idea when this will be merged and released?

+1, useful functionality, I would like to get information about the merge and release.

**Comment by LillaEli** [comment]
Any chances this could be merged in the near future! 🙏

**Comment by evgenyilin** [comment]
+1 When?)

**Comment by johannesdold** [comment]
+1 We would urgently need this option - any plans when this pr will be merged?

**Comment by rmertens** [comment]
we're also waiting for this. is there any news on this?

**Comment by alisman** [comment]
While it would be great for this PR to be merged, I believe I managed to accomplish this with the following hackery:

`onInputChange={(value: string) =>{
     this.selectRef.select.getNextFocusedOption = ()=>null;
}}`

There is no longer any default selection, but arrow keying still works.   

I then also bind to onKeyDown to intercept submission from input and fire submission callback with text directly from input.

**Comment by paulpiseckiy** [comment]
@alisman you have a great but incomplete solution here 🙂The problem is that it doesn't work when you open menu for the first time. Here's the complete:
```javascript
const ref = React.useRef();
  
React.useEffect(() => {
  ref.current.select.getNextFocusedOption = () => null;
}, []);

const handleInputChange = () => {
  ref.current.select.getNextFocusedOption = () => null;
};

const options = [
  {label: 'Option 1', value: '1'},
  {label: 'Option 2', value: '2'},
];

return (
  <ReactSelect
    ref={ref}
    options={options}
    onInputChange={handleInputChange}
  />
);
```

**Comment by bladey** [comment]
PR to be reviewed, duplicate of #3010

**Comment by bladey** [comment]
Hi all,

Thanks for your contribution @CharlieBreval!

There are a number of pull requests adding this feature - I've since created an up to date PR - #4080, and closed similar pull requests.

If I've missed anything, please let me know.

Closing this PR in favor of #4080.

**Pull request closed without merging**